src/helpers/randomizeCard/index.js

```diff
@@ -0,0 +1,736 @@
+			" Droplings",
+			" Visions",
+			" Spirits",
+			" Channelers"
+		]
+	],
+	["dwarf",
+		["Myst","Snow","Rock","Hearth","Wolf","Earth","Sleet","Chill","Hail","Frost","Ore","Shale","Stone"],
+		["wives","masons","workers","guards","cloaks","fathers","menders","stompers","beards","smiths","forgers","singers"]
+	],
+	["rodent",
+		["Ozone","Sound","Absorbing","Windy","Chaotic","Armed","Booming","Mechanical","Crazy"],
+		[" Lackeys"," Purifiers"," Drivers"," Varmints"," Minions"," Launchers"," Chargers"," Agents"," Pupils"," Schemers"," Professors"," Officers"," Workers"," Bombers"]
+	],
+	["construct",
+		["Function","Linked","Finite","Scrapped","Debug","Projected","Plated","Oxidized","Sleek","Automated","Hardwired","Boolean","Software","Motorized","Analog"],
+		[" Prototypes"," Wilds"," Golems"," Loopers"," Servers"," Planners"," Loggers"," Automatons"," Apparati"," Engines"]
+	],
+	["satyr",
+		["Lawless","Restless","Mindless","Pan","Wasteland","Desolate","Arid","Collective","Horned"],
+		[" Herd"," Goats"," Companions"," Horde"," Shepards"," Heralds"," Bucks"," Fawns"]
+	],
+	["undead",
+		["Forgotten","Shady","Mischevious","Petrified","Grim","Vindicative","Lasting"],
+		[" Regrets"," Souls"," Ghouls"," Liches"," Summoners"," Fossils"," Couriers"," Harvesters"," Martyrs"," Remains"]
+	],
+	["pirate",
+		["Northsea","Westwind","Cabin","Bluesail","Lucky","Seasick","Starboard"],
+		[" Dogs"," Mutineers"," Looters"," Sailors"," Privaters"," Captains"," Raiders"," Swindlers"," Charmers"," Bouncers"]
+	]
+]
+
+var statlines = [
+	[2,4,5,7,8,10,11,13], // 0 Speed
+	[1,2,3,5,6,8,9,11],	 // 1 Speed
+	[0,1,2,3,4,5,6,7],		// 2 Speed
+	[0,0,1,2,3,4,5,6]		 // 3 Speed
+]
+
+//###########################
+//# Code to generate a card #
+//###########################
+
+class Card {
+	constructor() {
+		this.mana = 0
+		this.effCost = 0
+		this.effCostMult = 0
+		this.str = 0
+		this.mov = 0
+		this.tribe = ""
+		this.subtribe = ""
+		this.faction = ""
+		this.effect = ""
+		this.rarity = ""
+		this.firstName = "Random"
+		this.lastName = "unit"
+	}
+
+	generate() {
+		while(true) {
+			this.effect = ""
+			this.effCostMult = 0
+			this.effCost = 0
+			this.subtribe = ""
+			
+			this.getTribe()
+			this.getEffect()
+			this.getStatline()
+			if(uncapped) {
+				maxEffCost = 999
+				minMana = -999
+				maxMana = 999
+			}
+			var sum = this.mana + this.effCost
+			var reroll = false
+			if (sum < minMana) {reroll = true}
+			if (sum > maxMana) {reroll = true} 
+			if (this.effCost > maxEffCost) {reroll = true}
+			if (sum > 6) {
+				if (this.effCost < 2) {reroll = true}
+			}
+			if ((this.trigger == 1) && (this.effect.includes('stronger') || this.effect.includes('weaker'))) {reroll = true}
+			if (reroll==false) {break}
+			if (verbose) {print("Rerolling effect\n")}
+		}
+		this.getName()
+	}
+
+	getTribe() {
+		this.tribe = rchoice(tribes)
+		this.faction = this.tribe.faction
+		if (faction != "" && this.faction != faction) {this.getTribe()} 
```

Comment:
We’d typically use strict checks.
```suggestion
		if (faction !== "" && this.faction !== faction) {this.getTribe()} 
```